### ISSUE TITLE
compatibility with theoldreader.com API

### ISSUE THREAD

**Issue by bor** [Alkarex/EasyRSS#15]
Hello,

Thanks for sharing a great app! Especially on f-droid repo.

I have tried to use it with theoldreader.com account.
And have some issue.
Login, initial sync seems like OK, but sync after (marked) read smth is fail.
I have got 2 kind of errors:
- Sync failed: Invalid HTTP status 404 ... /api/0/stream/contents/user/-/label/labelname?n=...
- Sync failed: Invalid JSON

I think a source of issues can be in a some [difference](https://github.com/theoldreader/api#differences) to Google Reader API.
There is an API link: https://github.com/theoldreader/api

Is it possible to fix those issues?

**Comment by Alkarex** [comment]
Yes, there are indeed a few differences between the original Google Reader API and TheOldReader. It should be possible to make a special case when detecting TheOldReader server without too many changes. I cannot promise to find time for that soon, but would welcome a patch.

**Comment by scastrorico** [comment]
I use this app for theoldreader.com with this change. Seemingly the problem was that theoldreader.com API use ObjectId type for identifiers, and app parse it to Long. This generate a NumberFormatException.

Using ObjectID string value as UIDs in app if this exception was thrown solved the problem.

`
--- a/src/org/freshrss/easyrss/data/parser/ItemIdJSONParser.java
+++ b/src/org/freshrss/easyrss/data/parser/ItemIdJSONParser.java
@@ -50,7 +50,12 @@ public class ItemIdJSONParser {
             case VALUE_STRING:
                 if (level == 3) {
                     if ("id".equals(name)) {
-                        itemId.setUid(Long.toHexString(Long.valueOf(parser.getText())));
+                    	String uid = parser.getText();
+                    	try {
+                    		uid = Long.toHexString(Long.valueOf(parser.getText()));
+                    	}
+                    	catch (final NumberFormatException nfe) {}
+                        itemId.setUid(uid);
                     } else if ("timestampUsec".equals(name)) {
                         itemId.setTimestamp(Long.valueOf(parser.getText()));
                     }

`

**Comment by Alkarex** [comment]
https://github.com/Alkarex/EasyRSS/pull/20

**Comment by Alkarex** [comment]
Thanks for the patch, @scastrorico

**Comment by Alkarex** [comment]
I have cherry-picked from @scastrorico patch https://github.com/Alkarex/EasyRSS/commit/c4f4a1be5390fe3c6237893850d0bad6f26103a2
